src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -89,10 +90,9 @@ internal WebAssemblyHostBuilder(IJSUnmarshalledRuntime jsRuntime, JsonSerializer
     }
 
     [UnconditionalSuppressMessage("Trimming", "IL2072", Justification = "Root components are expected to be defined in assemblies that do not get trimmed.")]
-    private void InitializeRegisteredRootComponents(IJSUnmarshalledRuntime jsRuntime)
+    private void InitializeRegisteredRootComponents(IInternalJSImportMethods jsMethods)
     {
-#pragma warning disable CS0618 // Type or member is obsolete
-        var componentsCount = jsRuntime.InvokeUnmarshalled<int>(RegisteredComponentsInterop.GetRegisteredComponentsCount);
+        var componentsCount = jsMethods.RegisteredComponents_GetRegisteredComponentsCount();
```

Comment:
I don't know if this is perf critical but if so, it could be re-implemented as passing `int[] Ids`, `string[] assemblies` etc.

Reply:
Good point. However it's only a one-time thing that happens during startup, so my guess is that we wouldn't be able to observe any difference.